lab09-kaylee/lib/storage.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+const storage = {};
+
+module.exports = exports = {};
+
+exports.createNote = function(schema, note) {
+  debug('#createNote');
+  if(!schema) return Promise.reject(new Error('schema required'));
+  if(!note) return Promise.reject(new Error('note required'));
+  if(!storage[schema]) storage[schema] = {};
+
+  storage[schema][note.id] = note;
+
+  let newNote;
+  return fs.writeFileProm(`${__dirname}/../data/${note.id}.json`, JSON.stringify(note))
+  .then(() => {
+    return fs.readFileProm(`${__dirname}/../data/${note.id}.json`)
+    .then(data => {
+      newNote = JSON.parse(data.toString());
+    })
+    .catch(err => console.error(err));
+  })
+  .then(() => {
+    return newNote;
+  })
+  .catch(err => {
+    console.error('error writing file in createNote', err.message);
+  });
+};
+
+exports.fetchNote = function(schema, id) {
+  debug('#fetchNote');
+
+  if(!schema) return Promise.reject(new Error('schema required'));
+  if(!id) return Promise.reject(new Error('id required'));
+
+  return fs.readFileProm(`${__dirname}/../data/${id}.json`)
+  .then(data => {
+    console.log('successfully reading file in fetchNote', data);
+    return JSON.parse(data.toString());
+  })
+  .catch(err => console.error('error reading file in fetchNote', err.message));
+};
+
+exports.updateNote = function(schema, id, newNote) {
```

Comment:
**TODO FOR LOST POINTS**

Update is buggy. Look at your ``createNote`` function and copy the same format of that because back in your ``PUT`` route, we only get back undefined so we're not returning the actual changed JSON object.